src/include/buffer.h

```diff
@@ -414,9 +415,14 @@ namespace buffer CEPH_BUFFER_API {
       _buffers.push_front(bp);
       _len += bp.length();
     }
+    void push_front(ptr&& bp) {
+      if (bp.length() == 0)
+	return;
+      _buffers.push_front(bp);
+      _len += bp.length();
```

Comment:
same, need to `std::move(bp)` when passing it to push_front(). then you'll need to call to bp.length() before it moves away:
```
_len += bp.length();
buffers.push_front(std::move(bp));
```